logic/src/commonMain/kotlin/com/wire/kalium/logic/data/asset/AssetRepository.kt

```diff
@@ -48,22 +84,28 @@ internal class AssetDataSource(
     }
 
     override suspend fun downloadPublicAsset(assetKey: String): Either<CoreFailure, ByteArray> = suspending {
-        wrapStorageRequest { assetDao.getAssetByKey(assetKey).firstOrNull() }
-            .coFold({
-                wrapApiRequest { assetApi.downloadAsset(assetKey, null) }
-                    .flatMap { assetData ->
-                        wrapStorageRequest { assetDao.insertAsset(assetMapper.fromUserAssetToDaoModel(assetKey, assetData)) }
-                            .map { assetData }
-                    }
-            }, {
-                Either.Right(it.rawData)
-            })
+        downloadAsset(assetKey, null)
+    }
 
+    override suspend fun downloadPrivateAsset(assetKey: String, assetToken: String): Either<CoreFailure, ByteArray> = suspending {
+        downloadAsset(assetKey, assetToken)
+    }
 
+    private suspend fun downloadAsset(assetKey: String, assetToken: String?): Either<CoreFailure, ByteArray> = suspending {
```

Comment:
I suggest that if we usually pass the assetToken as null, we can set the default value as null here!

Reply:
We don't know whether the assetToken will be null or not, as other clients (web) seem to not include it on the protobuff asset message. Therefore at this point we decided not to favor any preferred value of it but rather handle both cases